src/sessions/softphone-session-handler.ts

```diff
@@ -107,11 +107,36 @@ export class SoftphoneSessionHandler extends BaseSessionHandler {
   }
 
   async handlePropose (pendingSession: IPendingSession): Promise<void> {
-    await super.handlePropose(pendingSession);
+    const isPrivAnswerAuto = pendingSession.privAnswerMode === 'Auto';
+    const eagerConnectionEstablishmentMode = this.sdk._config.eagerPersistentConnectionEstablishment;
+    const logInfo = { sessionId: pendingSession?.id, conversationId: pendingSession.conversationId };
+    
+    if (isPrivAnswerAuto) {
+      this.log('info', 'received a propose with privAnswerMode=true', logInfo);
+    }
+    
+    // if eagerPersistentConnectionEstablishment==='none' then we want to completely swallow the propose
+    const shouldSwallowPropose = isPrivAnswerAuto && eagerConnectionEstablishmentMode === 'none';
```

Comment:
Is there a more elegant way to name this?

Reply:
done.